src/IssueViz.Security/Hotspots/LocalHotspotStoreMonitor.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.ConnectedMode.Hotspots;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Hotspots
+{
+    /// <summary>
+    /// Listens for changes the local hotspot store and sets the <see cref="LocalHotspotIssuesExistUIContext"/>
+    /// appropriately (which in turn will trigger showing/hiding the local hotspots tool window)
+    /// </summary>
+    [Export(typeof(ILocalHotspotStoreMonitor))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class LocalHotspotStoreMonitor : ILocalHotspotStoreMonitor, IDisposable
+    {
+        private readonly IServiceProvider serviceProvider;
+        private readonly ILocalHotspotsStore localHotspotsStore;
+        private readonly IThreadHandling threadHandling;
+
+        private IVsMonitorSelection vsMonitorSelection;
+        private uint contextCookie;
+
+        [ImportingConstructor]
+        public LocalHotspotStoreMonitor([Import(typeof(SVsServiceProvider))]IServiceProvider serviceProvider,
+            ILocalHotspotsStore localHotspotsStore,
+            IThreadHandling threadHandling)
+        {
+            this.serviceProvider = serviceProvider;
+            this.localHotspotsStore = localHotspotsStore;
+            this.threadHandling = threadHandling;
+        }
+
+        public async Task InitializeAsync()
+        {
+            localHotspotsStore.IssuesChanged += OnIssuesChanged;
+
+            await threadHandling.RunOnUIThread(() =>
+            {
+                vsMonitorSelection = (IVsMonitorSelection)serviceProvider.GetService(typeof(SVsShellMonitorSelection));
```

Comment:
This code is based on the implementation used to [show/hide the taint window](https://github.com/SonarSource/sonarlint-visualstudio/blob/7543d596f851c0d09b910da7996162d6c781531b/src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs#L81), although the thread handling is more correct in this version.